src/server/Common.re

```diff
@@ -11,6 +11,17 @@ let log: Middleware.t('a, 'a) =
     ctx.state;
   };
 
+module BodyParser = {
```

Comment:
Some thoughts below:

`json` should take a decoder, so that you can return a `state` that may have it. Ideally we could prove that your state has changed to include this, but that might be out of the scope of the first iteration. A signature like: `json : Middleware.t('a, 'b)` could be enough for now.

Should it be fixed on Yojson? Or should it just be a function that goes from `string => result('a, 'error)`? That way you can hook up Yojson but also use this on node with bs-decode or bs-json. I guess in that case this body-parser is a lot more general since you could also parse anything else.

What do you say if we deal with the body parser on a second PR? 😃

Reply:
Sure, it makes sense to me.

> I guess in that case this body-parser is a lot more general since you could also parse anything else

Right, so maybe it can just be a function like `Common.body_parser(...)` instead of a module.